server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -171,6 +172,31 @@ private static void startCliMonitorThread(InputStream stdin) {
         }).start();
     }
 
+    /**
+     * Writes the current process id into the given pidfile, if not null. The pidfile is cleaned up on system exit.
+     *
+     * @param pidFile A path to a file, or null of no pidfile should be written
+     */
+    private static void initPidFile(Path pidFile) throws IOException {
+        if (pidFile == null) {
+            return;
+        }
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            if (Files.exists(pidFile)) {
+                try {
+                    Files.delete(pidFile);
+                } catch (IOException e) {
+                    // ignore, nothing we can do because are shutting down
+                }
+            }
+        }, "elasticsearch[pidfile-cleanup]"));
```

Comment:
Good catch here to name this thread! I think there's one test in `SimpleThreadPoolIT` that might need to know about this name now.

Reply:
I think it's not necessary because the thread isn't started until shutdown.